controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -369,7 +368,7 @@ func (r *DSCInitializationReconciler) watchMonitoringSecretResource(a client.Obj
 	return nil
 }
 
-func (r *DSCInitializationReconciler) watchDSCResource(_ client.Object) []reconcile.Request {
+func (r *DSCInitializationReconciler) watchDSCResource(_ context.Context, _ client.Object) []reconcile.Request {
 	instanceList := &dscv1.DataScienceClusterList{}
 	if err := r.Client.List(context.TODO(), instanceList); err != nil {
```

Comment:
```suggestion
func (r *DSCInitializationReconciler) watchDSCResource(ctx context.Context, _ client.Object) []reconcile.Request {
	instanceList := &dscv1.DataScienceClusterList{}
	if err := r.Client.List(ctx, instanceList); err != nil {
```

I also wonder if cannot use ignored client here too instead of `r.Client` - maybe we can simplify further. Can be for a follow-up PR if applicable.

Reply:
let me leave this comment as open for following chore

Reply:
Ignore this comment. I was delusional and misread.... it was `client.Object` not `client.Client`